From 44260aa86bc3017d03046cac0cfe1e4d13a82e45 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Sun, 9 Feb 2025 23:34:29 +0100
Subject: [PATCH] bootctl: minor reordering of fields in output

Let's move the currently used/default/oneshot entry output after the
basic info about the boot loader itself, since conceptually these are
objects kinda "one level down" from the boot loader perspective. Hence,
let's *first* show all info about the boot loader itself before we
display the objects it manages.

This is just a trivial change in output, just swaps th elines for these
fields with the ones showing where the boot loader is installed.

(cherry picked from commit af5b961ad8f22be04f47c1c0e729b1e6fd78b423)
---
 src/bootctl/bootctl-status.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/bootctl/bootctl-status.c b/src/bootctl/bootctl-status.c
index 25c9f66a22..6f38b0f793 100644
--- a/src/bootctl/bootctl-status.c
+++ b/src/bootctl/bootctl-status.c
@@ -480,13 +480,6 @@ int verb_status(int argc, char *argv[], void *userdata) {
                         (void) efi_loader_get_device_part_uuid(&loader_partition_uuid);
                         print_yes_no_line(/* first= */ false, !sd_id128_is_null(loader_partition_uuid), "Boot loader set partition information");
 
-                        if (current_entry)
-                                printf("Current Entry: %s\n", current_entry);
-                        if (default_entry)
-                                printf("Default Entry: %s\n", default_entry);
-                        if (oneshot_entry && !streq_ptr(oneshot_entry, default_entry))
-                                printf("OneShot Entry: %s\n", oneshot_entry);
-
                         if (!sd_id128_is_null(loader_partition_uuid)) {
                                 if (!sd_id128_is_null(esp_uuid) && !sd_id128_equal(esp_uuid, loader_partition_uuid))
                                         printf("WARNING: The boot loader reports a different partition UUID than the detected ESP ("SD_ID128_UUID_FORMAT_STR" vs. "SD_ID128_UUID_FORMAT_STR")!\n",
@@ -499,6 +492,14 @@ int verb_status(int argc, char *argv[], void *userdata) {
 
                         if (loader_path)
                                 printf("       Loader: %s%s\n", special_glyph(SPECIAL_GLYPH_TREE_RIGHT), strna(loader_path));
+
+                        if (current_entry)
+                                printf("Current Entry: %s\n", current_entry);
+                        if (default_entry)
+                                printf("Default Entry: %s\n", default_entry);
+                        if (oneshot_entry && !streq_ptr(oneshot_entry, default_entry))
+                                printf("OneShot Entry: %s\n", oneshot_entry);
+
                         printf("\n");
                 }
 
